Sources/HTTP/Serializer/RequestSerializer.swift

```diff
@@ -0,0 +1,72 @@
+import Transport
+
+public final class RequestSerializer<Stream: WriteableStream>: Serializer {
+    typealias StreamType = Stream
+    let stream: Stream
+    public init(_ stream: Stream) {
+        self.stream = stream
+    }
+    
+    public func serialize(_ request: Request) throws {
+        let message = try serialize(request as Message)
+        
+        let methodBytes: Bytes
+        switch request.method {
+        case .connect:
+            methodBytes = [.C, .O, .N, .N, .E, .C, .T]
+        case .delete:
+            methodBytes = [.D, .E, .L, .E, .T, .E]
+        case .get:
+            methodBytes = [.G, .E, .T]
+        case .head:
+            methodBytes = [.H, .E, .A, .D]
+        case .options:
+            methodBytes = [.O, .P, .T, .I, .O, .N, .S]
+        case .put:
+            methodBytes = [.P, .U, .T]
+        case .patch:
+            methodBytes = [.P, .A, .T, .C, .H]
+        case .post:
+            methodBytes = [.P, .O, .S, .T]
+        case .trace:
+            methodBytes = [.T, .R, .A, .C, .E]
+        case .other(let string):
+            methodBytes = string.makeBytes()
+        }
+        
+        var length = methodBytes.count
+            + 1 //space
+            + request.uri.path.characters.count
+        if let query = request.uri.query {
+            length += 1 + query.characters.count
+        }
+        length += 11 // _HTTP/1.1\r\n
+        length += message.count
+        
+        var buffer = Bytes()
+        buffer.reserveCapacity(length)
+        
+        buffer += methodBytes
+        buffer += [.space]
+        buffer += request.uri.path.makeBytes()
+        if let query = request.uri.query {
+            buffer += [.questionMark] + query.bytes
+        }
+        buffer += [.space]
+        buffer += [.H, .T, .T, .P, .forwardSlash, .one, .period, .one]
```

Comment:
Even if you modify the version of HTTP, this is still an HTTP/1.1 serializer. Unless we update, we won't be compatible with things from 1.2 that aren't backward compatible with 1.1. Given the fairly large performance hit we take from serializing the version that comes from the message, I don't think it's worth it.